server/chat-commands.js

```diff
@@ -3796,6 +3796,49 @@ const commands = {
 	},
 	importinputloghelp: [`/importinputlog [inputlog] - Starts a battle with a given inputlog. Requires: + % @ & ~`],
 
+	acceptdraw: 'accepttie',
+	offerdraw: 'offertie',
+	accepttie: 'offertie',
+	offertie(target, room, user, connection, cmd) {
+		const battle = room.battle;
+		if (!battle) return this.errorReply("Must be in a battle room.");
+		if (!Config.allowrequestingties) {
+			return this.errorReply("This server does not allow offering ties.");
+		}
+		if (!this.can('roomvoice', null, room)) return;
+		if (cmd === 'accepttie' && !battle.players.some(player => player.wantsTie)) {
+			return this.errorReply("No tie offer available. It might have been automatically withdrawn as the current turn started.");
```

Comment:
"automatically withdrawn" is unnecessarily complicated phrasing.

Reply:
And also not really the core of the problem which is that the opponent took back the offer.

Reply:
```suggestion
			return this.errorReply("No other player is requesting a tie right now. It was probably canceled.");
```

Reply:
Actually, I still don't believe this needs to exist at all. Maybe the correct solution is that draw offers should cancel and prevent moves from being made.

Reply:
No, the timer still exists. Rather, I think draw offers should last across turns, instead of needing reconfirming every turn.

Reply:
Or maybe something weird, like lasting exactly two turns.

Reply:
I feel like that's a discussion for Policy Review, so we should go with the safe option for now which is resetting after every turn.